api/notd/metadata_manager.py

```diff
@@ -0,0 +1,38 @@
+import json
+import logging
+import os
+import urllib.request
+
+from web3 import Web3
+
+from notd.model import TokenMetadata
+
+_EMPTY_STRING = '_EMPTY_STRING'
+w3 = Web3(Web3.HTTPProvider(f'https://mainnet.infura.io/v3/{os.environ["INFURA_PROJECT_ID"]}'))
+
+def retrieve_token_metadata(registryAddress: str,tokenId:int):
+    with open('./contracts/IERC721Metadata.json') as contractJsonFile:
+        erc721MetdataContractJson = json.load(contractJsonFile)
+        erc721MetdataContractAbi = erc721MetdataContractJson['abi']
+    contract = w3.eth.contract(registryAddress,abi=erc721MetdataContractAbi)
+    url = contract.functions.tokenURI(tokenId).call()
+    with urllib.request.urlopen(url) as response:
+        data = json.loads(response.read())
+        metadataUrl = url
+        imageUrl = data.get('image',_EMPTY_STRING)
```

Comment:
dont do empty string stuff here, thats DB only. This class at the moment doesnt not interact with the db at all so should not have any DB related stuff. an easy way to tell is there should be no imports from store. Even when we do this later _EMPTY_STRING is very internal to the Saver class, it should never be used outside. It's set as the default parameter value (read up on these if thats unclear) so it gets set by python if the param hasnt got a value specified in the call

Reply:
if the object is not present in the json , it throws and error, was thinking of just returning an empty value instead